### PR TITLE
174 prepended appended inputs don’t respect width modifiers and aren't responsive

### PR DESCRIPTION
**Pull request by jamesjacobs**
input groups

### CODE REVIEW

File: stylesheets/_component.input-groups.scss [hunk 1]
```diff
@@ -5,17 +5,14 @@
 // Base styles
 // -------------------------
 .input-group {
-    position: relative; // For dropdowns
-    display: table;
-    height: 40px;
     border-collapse: separate; // prevent input groups from inheriting border styles from table cells when placed within a table
+    display: table;
+    position: relative;
+    position: relative; // For dropdowns
```
Comment (langy) [inline stylesheets/_component.input-groups.scss]: Duplicates line immediately above

File: stylesheets/_component.input-groups.scss [hunk 2]
```diff
@@ -24,12 +21,9 @@
...
     .form__control {
-        width: 75%;
-        margin-bottom: 0;
-
-        @include respond-min($screen-phone) {
-            width: 100%;
-        }
+        display: table-cell;
+        position: relative;
+        float: left;
```
Comment (langy) [inline stylesheets/_component.input-groups.scss]: Alphabetise properties

**Comment by langy** [comment]
👍

File: stylesheets/_component.input-groups.scss (hunk 2 above)
Comment (jamesjacobs) [inline stylesheets/_component.input-groups.scss]: Bah, forgot the linter doesn't cover this yet.

**Comment by jamesjacobs** [comment]
Added support for appended/prepended buttons

input groups with buttons

File: stylesheets/_component.input-groups.scss [hunk 3]
```diff
@@ -38,28 +32,58 @@
...
     &:not(:first-child):not(:last-child) {
         border-radius: 0;
     }
 }
-// Addon and addon wrapper for buttons
-.input-group-addon,
+
+.input-group-addon {
+    background-color: color(gray, lightest);
+    border-radius: $border-radius;
```
Comment (langy) [inline stylesheets/_component.input-groups.scss]: Move this below the border for alphabeticallyness

File: stylesheets/_component.input-groups.scss [hunk 4]
```diff
@@ -38,28 +32,58 @@
...
 }
 
+
 // Text input groups
 // -------------------------
 .input-group-addon,
-.input-group-guidance {
+.input-group-guidance,
+.input-group-btn .btn {
     padding: 0 $padding-base-horizontal;
```
Comment (langy) [inline stylesheets/_component.input-groups.scss]: Alphabetise (although aware this wasn't something this PR introduced)

File: stylesheets/_component.input-groups.scss [hunk 5]
```diff
@@ -73,11 +97,7 @@
     }
 }
 
-.input-group-addon {
-    background-color: color(gray, lightest);
-    border: 1px solid $input-group-addon-border-color;
-    border-radius: $border-radius;
-}
+// !! borders on appended buttons aren't right!!!
```
Comment (langy) [inline stylesheets/_component.input-groups.scss]: Does this need to remain?

**Comment by langy** [comment]
👍

**Comment by jamesjacobs** [comment]
Closes #174

**Pull request merged**